src/posix/threading.c

```diff
@@ -195,3 +223,154 @@ void sys_thread_join(sys_thread_t *thread)
     assert(thread != &_main_thread_id);
     pthread_join(thread->thread, 0);
 }
+
+sys_thread_role_t sys_thread_role(const sys_thread_t *thread)
+{
+    if (!thread)
+        thread = sys_thread_self();
+    return thread->role;
+}
+
+sys_thread_proactor_mode_t sys_thread_proactor_mode(void)
```

Comment:
I see a function called `sys_thread_proactor_set_mode()`, so should this function be called 'sys_thread_proactor_get_mode()' ?

Reply:
I was trying to keep it consistent with the other simple accessor functions, namely sys_thread_name() and sys_thread_self(), for with the "get" is implied.